lhotse/dataset/fields.py

```diff
@@ -0,0 +1,338 @@
+                frame_shift=cut.frame_shift,
+                # Note: Rounding "floor" can sometimes result in one extra frame being included
+                # in the left context; but it guarantees that we will never go out-of-bounds when
+                # summing start_frame + num_frames.
+                rounding=ROUND_FLOOR
+            ),
+            'num_frames': compute_num_frames(
+                supervision.duration,
+                frame_shift=cut.frame_shift
+            )
+        }
+
+
+class AudioSpan:
+    """
+    Returns a field describing start samples and number of samples for the supervision.
+
+    Output:
+
+    .. code-block:
+
+        output = {'start_sample': int, 'num_samples': int}
+    """
+
+    def __call__(self, supervision: SupervisionSegment, cut: AnyCut) -> Dict[str, Any]:
+        return {
+            'start_sample': compute_num_samples(
+                supervision.start,
+                sampling_rate=cut.sampling_rate,
+                rounding=ROUND_FLOOR
+            ),
+            'num_samples': compute_num_samples(
+                supervision.duration,
+                sampling_rate=cut.sampling_rate
+            )
+        }
+
+
+class Text:
+    """
+    Returns a field with the transcription text (as string).
+
+    Output:
+
+    .. code-block:
+
+        output = {'text': str}
+    """
+
+    def __call__(self, supervision: SupervisionSegment, cut: AnyCut) -> Dict[str, Any]:
+        return {'text': supervision.text}
+
+
+class Characters:
+    """
+    Returns a field with the transcription text as a sequence of (string) characters.
+    During collation, it will add `<PAD>` symbols at the end to match the longest
+    sequence in the batch.
+
+    Output:
+
+    .. code-block:
+
+        output = {'chars': List[str], 'chars_len': int}
+    """
+
+    def __init__(self):
+        self.pad = '<PAD>'
+
+    def __call__(self, supervision: SupervisionSegment, cut: AnyCut) -> Dict[str, Any]:
+        return {'chars': list(supervision.text), 'chars_len': len(supervision.text)}
+
+    def collate(self, items: List[str], key: str):
+        assert key in ('chars', 'chars_len')
+        if key == 'chars':
+            items = sorted(items, key=len, reverse=True)
+            for idx in range(1, len(items)):
+                items[idx].extend([self.pad] * (len(items[0]) - len(items[idx])))
+        elif key == 'chars_len':
+            items = default_collate(items)
+        return items
+
+
+class CharacterIds:
+    """
+    Returns a field with the transcription text as a sequence of (int) character ids.
+    During collation, it will add `<PAD>` symbol ids at the end to match the longest
+    sequence in the batch.
+
+    Output:
+
+    .. code-block:
+
+        output = {'char_ids': List[int], 'char_ids_len': int}
+    """
+
+    def __init__(self, cuts):
+        self.pad = '<PAD>'
+        self.id2sym = sorted({char for cut in cuts for s in cut.supervisions for char in s.text}) + [self.pad]
```

Comment:
Mm, it seems to me a little problematic to set the ids2sym and sym2id in this way (this is essentially a symbol table, I think; and in k2 we'd probably be wanting to load it from disk...)
This makes it a function of the dataset, but sometimes there are rare characters that might not appear in a given collection of text, or characters that should be considered as OOV.